azurerm/internal/services/datamigration/resource_arm_data_migration_service.go

```diff
@@ -0,0 +1,241 @@
+	return &schema.Resource{
+		Create: resourceArmDataMigrationServiceCreate,
+		Read:   resourceArmDataMigrationServiceRead,
+		Update: resourceArmDataMigrationServiceUpdate,
+		Delete: resourceArmDataMigrationServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					// No const defined in go sdk, the literal listed below is derived from the response of listskus endpoint.
+					// See: https://docs.microsoft.com/en-us/rest/api/datamigration/resourceskus/listskus
+					"Premium_4vCores",
+					"Standard_1vCores",
+					"Standard_2vCores",
+					"Standard_4vCores",
+				}, false),
+			},
+
+			"kind": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "Cloud",
+				ValidateFunc: validation.StringInSlice([]string{"Cloud"}, false),
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmDataMigrationServiceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataMigration.ServicesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Data Migration Service (Service Name %q / Group Name %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_migration_service", *existing.ID)
+		}
+	}
+
+	skuName := d.Get("sku_name").(string)
+	subnetID := d.Get("subnet_id").(string)
+	location := d.Get("location").(string)
+
+	parameters := datamigration.Service{
+		Location: utils.String(location),
+		ServiceProperties: &datamigration.ServiceProperties{
+			VirtualSubnetID: utils.String(subnetID),
+		},
+		Sku: &datamigration.ServiceSku{Name: utils.String(skuName)},
```

Comment:
_minor_ could we new-line this for readability?

```suggestion
		Sku: &datamigration.ServiceSku{
			Name: utils.String(skuName),
		},
```

Reply:
Sure